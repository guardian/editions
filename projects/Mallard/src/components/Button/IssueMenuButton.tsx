import React from 'react';
import { Button } from './Button';

const IssueMenuButton = ({ onPress }: { onPress: () => void }) => (
	<Button icon={'\uE04A'} alt="More issues" onPress={onPress} />
);

export { IssueMenuButton };
