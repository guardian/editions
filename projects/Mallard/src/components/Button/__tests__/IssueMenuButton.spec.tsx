import React from 'react';
import type { ReactTestRendererJSON } from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
import { IssueMenuButton } from '../IssueMenuButton';

describe('IssueMenuButton', () => {
	it('should display a IssueMenuButton', () => {
		const component: ReactTestRendererJSON | null = TestRenderer.create(
			<IssueMenuButton onPress={() => {}} />,
		).toJSON();
		expect(component).toMatchSnapshot();
	});
});
