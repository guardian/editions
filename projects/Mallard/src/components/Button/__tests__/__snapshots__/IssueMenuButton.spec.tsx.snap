// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueMenuButton should display a IssueMenuButton 1`] = `
<View
  accessibilityHint="More issues"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 999,
          "height": 42,
          "justifyContent": "center",
          "paddingHorizontal": 21,
        },
        {
          "backgroundColor": "#ffe500",
        },
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "justifyContent": "center",
          "paddingHorizontal": 0,
          "width": 42,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        [
          {
            "fontFamily": "GuardianIcons-Regular",
            "fontSize": 20,
            "lineHeight": 20,
          },
          [
            {
              "color": "#121212",
            },
            undefined,
          ],
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
